reactor-core/src/test/java/reactor/core/publisher/SinkEmptyMulticastTest.java

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright (c) 2011-Present VMware Inc. or its affiliates, All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *        https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package reactor.core.publisher;
+
+import java.time.Duration;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicReference;
+
+import org.junit.jupiter.api.Test;
+import org.reactivestreams.Subscriber;
+
+import reactor.core.CoreSubscriber;
+import reactor.core.Scannable;
+import reactor.core.scheduler.Schedulers;
+import reactor.test.StepVerifier;
+
+import static org.assertj.core.api.Assertions.*;
+import static reactor.core.Scannable.Attr.RUN_STYLE;
+import static reactor.core.Scannable.Attr.RunStyle.SYNC;
+
+class SinkEmptyMulticastTest {
+
+	@Test
+	void currentSubscriberCount() {
+		Sinks.Empty<Integer> sink = new SinkEmptyMulticast<>();
+
+		assertThat(sink.currentSubscriberCount()).isZero();
+
+		sink.asMono().subscribe();
+
+		assertThat(sink.currentSubscriberCount()).isOne();
+
+		sink.asMono().subscribe();
+
+		assertThat(sink.currentSubscriberCount()).isEqualTo(2);
+	}
+
+	@Test
+	void currentSubscriberCountReflectsCancellation() {
+		SinkEmptyMulticast<Void> mp = new SinkEmptyMulticast<>();
+		StepVerifier.create(mp)
+				.thenCancel()
+				.verify();
+
+		assertThat(mp.currentSubscriberCount()).isEqualTo(0);
+	}
+
+	@Test
+	void blockReturnsNullOnTryEmitEmpty() {
+		SinkEmptyMulticast<Void> mp = new SinkEmptyMulticast<>();
+
+		Schedulers.parallel().schedule(() -> mp.tryEmitEmpty(), 50L, TimeUnit.MILLISECONDS);
+		assertThat(mp.block(Duration.ofSeconds(1))).isNull();
```

Comment:
why not `StepVerifier`?

Reply:
this is testing that the behavior of `block()` is consistent, especially since the processor-specific implementation was removed in the commit.

Reply:
perhaps it should be a dedicated test then? Not a big deal but I was first confused to see `block()` used given that it is an empty processor that should not emit anything, and only then I noticed `isNull()`, so it is _a bit_ counterintuitive when you read the test :)

Reply:
test is named `blockReturnsNullOnTryEmitEmpty`... 😕

Reply:
ooops, sorry, I missed that! Although then we miss tests for "normal" subscribe on `asMono()` :)

Reply:
ah, yeah we tried to improve the test coverage but that fell through I guess

Reply:
test added :)